actionview/lib/action_view/base.rb

```diff
@@ -162,6 +162,9 @@ class Base
     # Annotate rendered view with file names
     cattr_accessor :annotate_rendered_view_with_filenames, default: false
 
+    # Specify whether an error should be raised for missing translations
+    cattr_accessor :raise_on_missing_translations, default: false
```

Comment:
I think this should be on `ActionView` itself, as it's a pattern we started to use to stop using class variables. 

```ruby
module ActionView
  singleton_class.attr_accessor :raise_on_missing_translations
  self.raise_on_missing_translations = false
end

```

Reply:
I can do this, but then we have the same problem. `ActiveSupport.on_load(:action_view)` does not fire for `ActionView` or its properties, only `ActionView::Base`.

Reply:
I may be wrong, but I think you just wouldn't need the `on_load` at all anymore.

Reply:
I don't think you don't need a hook, a regular initializer should do. Or am I missing something?

Reply:
Oh you mean?
```ruby
module ActionView
  class Railtie < Rails::Engine 
    config.after_initialize do |app|
      ActionView.raise_on_missing_translations = app.config.i18n.raise_on_missing_translations 
    end
```

Could work. Let me validate.

Reply:
No, it does not appear that that would work.

The railtie for `i18n` is in active support. So it goes:
* Load `active_support`
* This loads the `i18n_railtie`
* This triggers `config.after_initialize do |app|  I18n::Railtie.initialize_i18n(app)` [here](https://github.com/rails/rails/blob/51b4370bb371a0f1d0823c57cad3a1557bf1e6fe/activesupport/lib/active_support/i18n_railtie.rb#L17-L21)
* This adds the `ActiveSupport.on_load(:action_view)` to copy `raise_on_missing_translations`

So it's not actually an `ActionView` initializer that copies over this config.

I might be missing something too, though.

Reply:
Looking at it another way, because the i18n railtie exists in active support, we need an on_load hook for cases where we use active support and not action view. And (I think) the only appropriate on_load hook we have for action view is `ActionView::Base`.


We could move the assignment to the action view railtie, as long as it is safe to assume `app.config.i18n` will be defined, but I think it is easier to reason about if we leave it in the i18n railtie.

Reply:
Even then, the railtie doesn't assume the `raise_on_missing_translations` key exists on the config within `app.config.i18n`. It iterates over and assigns if it finds the key only.

Reply:
Can we move the setting to the singleton class though?